src/libs/core/src/compiler.cpp

```diff
@@ -6563,8 +6564,9 @@ bool COMPILER::SaveState(std::fstream &fileS)
     {
         char *pDst = new char[dwCurPointer * 2];
         uint32_t dwPackLen = dwCurPointer * 2;
+        uLongf tmpLong = dwPackLen;
         RDTSC_B(dw2);
-        compress2((Bytef *)pDst, (uLongf *)&dwPackLen, (Bytef *)pBuffer, dwCurPointer, Z_BEST_COMPRESSION);
+        compress2((Bytef *)pDst, &tmpLong, (Bytef *)pBuffer, dwCurPointer, Z_BEST_COMPRESSION);
```

Comment:
the output is written to tmpLong but dwPackLen is used afterwards.
this should write to dwPackLen instead

Reply:
Fixed.